mautic/mautic
# Fix autofill boolean fields

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/6096
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Reported issue https://github.com/mautic/mautic/issues/6096
Noticed wrong value sanitize for boolean fields.
Also fixed for mapping autofill boolean radio/checkbox group.
This PR fixed it

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create 2 boolean custom fields (like below)
![image](https://user-images.githubusercontent.com/31535432/40169232-d0917d26-59c5-11e8-86e8-1c4456ccf0ff.png)
2. Create a form and add 3 form field
2.1 Add email form field map to email custom field
2.2 Add a radio group form field map to the first boolean custom field previously created.
In the properties tab, click on YES
![image](https://user-images.githubusercontent.com/31535432/40169328-0fa3ae8a-59c6-11e8-88d8-32f72a86ee4b.png)
Go on the behavior tab and click on YES to auto fill data
![image](https://user-images.githubusercontent.com/31535432/40169356-1f2f271c-59c6-11e8-8714-ea7a7a9a5786.png)
2.3 do the same for the second radio group form field
![image](https://user-images.githubusercontent.com/31535432/40169391-30568fee-59c6-11e8-9d66-fd7386f58e60.png)
3. Create a landing page and add the form inside (with token or automatic copy, it's up to you)
4. Save the LP and open a new session.
5. Fill the form.
6. After filling the form, refresh the page and see that the first boolean field is checked on YES or NO (depends on what you filled).
See that the second boolean field is not checked (even if when you set, you clicked on auto fill data YES)
![image](https://user-images.githubusercontent.com/31535432/40169557-9074a9d8-59c6-11e8-9b8b-ea3e4053683e.png)
8. Go back in the form edition, change the order of boolean field
First I got the the bool1 and bool2
Now I have bool2 and bool1
![image](https://user-images.githubusercontent.com/31535432/40169621-b0d9a66a-59c6-11e8-8e1e-28a3bc7381ce.png)
Go back on the landing page and refresh
See that now bool2 is checked (earlier it was bool1) and the other one is not checked.
![image](https://user-images.githubusercontent.com/31535432/40169669-ccabe3da-59c6-11e8-93ba-4245545c018e.png)


#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Check If checkbox group/radio groups works properly for boolean type and autofill
